### ISSUE TITLE
Identify graph roots in the ProjectGraph

### ISSUE THREAD

**Issue by dfederm** [Microsoft/msbuild#3796]
Exposing graph roots (which may not necessarily be the same as the entry points when multiple are supported, see #3795) can help consumers identify when there are disconnected graphs in the ProjectGraph.

This is a "nice to have" expressed by some potential consumers. This work is only useful after #3795 is done.

**Comment by rainersigwald** [comment]
So this is the subset of entry points that aren't transitively reachable from other entry points? In this graph, it'd be just `Entry1`?

![image](https://user-images.githubusercontent.com/3347530/46095036-2d509d80-c181-11e8-89d8-a57ba7d18ba8.png)

**Comment by dfederm** [comment]
@rainersigwald exactly. Nodes with no incoming edges. If we had the Referenc**ing**Projects it'd be `AllNodes.Where(node => node.ReferencingProjects.Count == 0)` (but obviously not implemented that way)
